### DISCUSSION TITLE
6.0.3 - How does one add a new section to an existing Document

### DISCUSSION THREAD

**Discussion by gjblajian** [dolanmiu/docx#901] [Q&A] [answered]
With the switch to the declarative style for sections, how does one add a new section to an existing Document

**Comment by dolanmiu** [comment, marked as answer]
Start to think of `docx` like `HTML` then it starts making more sense

With `HTML`, you generally don't write html, then later, "add more `<div>`s to the html"

A html page is static, just like a Word document

So now, with a declarative style, you would take the data, then do a transform, then add it in `sections` all in one go

So theres three steps:

1. Prepare your all data fully
2. Pass the data into a new function to create a `Document`
3. Use `Packer` to export the document

Because these steps are now succinct with clear boundaries, there will be less bugs and will be easy to test

Adding sections (aka mutating the `Document`) blurs the lines between data and Word Document. It will be hard to unit test at the very least
